src/interfaces/books.ts

```diff
@@ -0,0 +1,14 @@
+import { Category, Tags } from '../types/books'
+
+export interface IBook {
+    id: number,
+    title: string,
+    author: string,
+    publisher?: string,
+    publication_date?: string,
```

Comment:
Nit - it would be good to choose a single naming convention for the project - either camelCase, or unix_hacker_style (or anything else you fancy), not both of them
Nit - should this be typed as a `Date` instead?

Reply:
Changing it to camelCase.
For Date, the JSON has year format, for e.g., "1954", so I'll prefer to keep it as string, as it's less complicated.

Reply:
Well, I guess it is not really a date, it is just a year, so maybe it should be a number
https://stackoverflow.com/a/25126647

Reply:
Date also has something like this, "8th century BCE". So I suppose this should remain a string?

Reply:
Oy. :| We can revisit, yuck